assets/component-facets.css

```diff
@@ -352,24 +450,32 @@
 }
 
 /* Hover, active, and focus states */
-:is(.facets__label:hover, .facets__label.active, .facets__label:has(:focus-visible)) {
+.facets__label:hover,
+.facets__label.active,
+.facets__label:has(:focus-visible) {
   color: rgba(var(--color-foreground), 1);
 }
 
-/* Focus states for older browsers */
-@supports not selector(:has(a, b)) {
-  .facets__label:focus-within {
-    color: rgba(var(--color-foreground), 1);
-  }
+/* Disabled state */
+.facets-layout .facets__label.disabled {
+  pointer-events: none;
 }
 
-/* Disabled state */
-.facets__label.disabled {
+.facets-layout:not(.facets-layout-list--swatch, .facets-layout-grid--image) .facets__label.disabled {
+  opacity: 0.4;
+}
+
+.facets-layout-grid--image .facets__label.disabled .facets__image-wrapper {
+  opacity: 0.2;
+}
+
+:is(.facets-layout-list--swatch, .facets-layout-grid--image) .facets__label.disabled .facet-checkbox__text {
   opacity: 0.4;
```

Comment:
> but for plain text I can't tell the difference.

I'm not personally bothered by this because I'm a firm believer of "don't rely on color alone to convey meaning". There are a number of good resources on this, but I'll point to the [WCAG one](https://www.w3.org/TR/WCAG20-TECHS/G14.html) and [this blog](https://www.accessibility-developer-guide.com/knowledge/colours-and-contrast/colour-is-not-enough/)

The product option selector use this color, but also strike through the options to convey that they are not options. These filter values communicate something similar by having a `(0)` beside it. So I wouldn't rely on the color contrast between 2 items to convey meaning; it's more important that the constrast between the foreground and background ensure legibility

Other thing: When I visit my parent's house and see their sites on their old laptop, these "low opacity" text is barely visible at all on their monitors.

Reply:
I forgot we do that 0.75 text opacity thing throughout Dawn. 🙃 I agree that level of difference isn't enough. Any changes we make here aren't just the swatches, but also the checkboxes, which have a less obvious visual difference between disabled and enabled.  

I'm inclined to just not touch this in this PR — this issue existed before we got here, and we can address it separately if we decide to do so. Either way, I don't think this is critical, personally. WCAG guidelines specifically [carve out an exception](https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html) for "Text or images of text that are part of an inactive user interface component" in their contrast guidelines.

Reply:
I agree with @kjellr . I found another part of the [WCAG spec](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html) having an exemption for disabled states:
![image](https://github.com/Shopify/dawn/assets/1019769/e04fac1a-b75a-4d4c-998c-559a55a5c0ed)

It's hard to know 100% if this qualifies as a "User Interface Component" - the checkbox certainly is, and I assume that includes the label, but not 100% sure.

What do you think about leaving as-is, since this is what it's been for a while, and it sounds like disabled elements are exempt from this?

Reply:
For what it's worth, Chrome's default user styles for a disabled button don't meet WCAG contrast.

![image](https://github.com/Shopify/dawn/assets/1019769/cc746657-5073-4781-9f3d-ba4a240c3e3a)

I think contrast is thrown out the window for disabled elements, so I think we should lean more towards UX guidance than contrast for this scenario. Are you okay with leaving as is?

Reply:
Sure, we can leave as is. I'm really confused by this guidance; I don't understand the written rationale and I think it's a poor experience for people who have trouble with contrast differences or folks with screen resolutions not as dense as ours, but I'll drop it.

Reply:
As for Chrome: the user-agent styles don't carry any weight for me. If chrome has a poor experience, that's on them, and we don't need to adopt it. Same thing for firefox's styles back in the day - they were barely noticeable so worth overwriting.